_posts/2025-11-12-agentic-ai-patterns.adoc

```diff
@@ -0,0 +1,182 @@
+---
+layout: post
+title: 'Agentic AI Patterns: one size does not fit all'
+date: 2025-11-12T00:00:00Z
+tags: ai llm agents
+synopsis: 'Agentic AI Patterns: one size does not fit all'
+author: mariofusco
+---
+:imagesdir: /assets/images/posts/agentic
+
+In the first months of 2025 I published a three-part blog post series on agentic AI with Quarkus and its LangChain4j extension where I first https://quarkus.io/blog/agentic-ai-with-quarkus/[introduced agentic AI and workflow patterns], then explored https://quarkus.io/blog/agentic-ai-with-quarkus-p2/[purely AI orchestrated agentic patterns], and finally examined the https://quarkus.io/blog/agentic-ai-with-quarkus-p3/[differences between these two approaches] with a practical example, trying to put in evidence their pros and cons.
+
+Only a handful of months are passed since those articles, but in the meantime the AI ecosystem, and the possibilities it enables, evolved at an incredible speed, and I'm trying hard to keep up with all the novelties and changes. In fact the initial experiments that I discussed in that series seem now to come from a different geological era, even though they served me as an important starting point to explore the agentic AI landscape.
```

Comment:
```suggestion
Only a handful of months have passed since those articles, but in the meantime the AI ecosystem, and the possibilities it enables, have evolved at an incredible speed, and I'm trying hard to keep up with all the novelties and changes. In fact the initial experiments that I discussed in that series seem now to come from a different geological era, even though they served me as an important starting point to explore the agentic AI landscape.
```